source/Private/Test-FabricApiResponse.ps1

```diff
@@ -0,0 +1,129 @@
+
+.PARAMETER ObjectIdOrName
+    The name or ID of the resource being operated.
+
+.PARAMETER TypeName
+    The type of resource being operated (default: 'Fabric Item').
+
+.PARAMETER NoWait
+    If specified, the function will not wait for the operation to complete and will return immediately.
+
+.EXAMPLE
+    Test-FabricApiResponse -statusCode 201 -response $response -responseHeader $header -Name "MyResource" -typeName "Fabric Item"
+
+    Handles the response from a Fabric API call with a 201 status code, indicating successful creation of a resource.
+
+.EXAMPLE
+    Test-FabricApiResponse -statusCode 202 -response $response -responseHeader $header -Name "MyResource" -typeName "Fabric Item"
+
+    Handles the response from a Fabric API call with a 202 status code, indicating that the request has been accepted for processing.
+
+.NOTES
+    - This function is designed to be used within the context of a Fabric API client.
+    - It requires the `Write-Message` function to log messages at different levels (Info, Debug, Error).
+    - The function handles long-running operations by checking the status of the operation and retrieving the result if it has succeeded.
+
+    Author: Kamil Nowinski
+
+#>
+
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $false)]
+        $Response,
+
+        [Parameter(Mandatory = $false)]
+        [string] $ObjectIdOrName,
+
+        [Parameter(Mandatory = $false)]
+        [string] $TypeName = 'Fabric Item',
+
+        [Parameter(Mandatory = $false)]
+        [switch] $NoWait = $false
+    )
+
+    $responseHeader = $script:responseHeader
+    $statusCode = $script:statusCode
+    $result = $null
+
+    $verb = (Get-PSCallStack)[1].Command.Split('-')[0]
+    Write-Message -Message "Testing API response for '$verb' operation. StatusCode: $statusCode." -Level Debug
+
+    switch ($statusCode) {
+        200 {
+            $result = $null
+        }
+        201 {
+            $result = $Response
+        }
+        202 {
+            Write-Message -Message "$verb Request for $TypeName '$ObjectIdOrName' accepted. Provisioning in progress!" -Level Info
+            [string]$operationId = $responseHeader["x-ms-operation-id"]
+
+            if ($NoWait) {
+                Write-Message -Message "NoWait parameter is set. Operation ID: $operationId" -Level Info
+                Write-Message -Message "Run to check the progress: Get-FabricLongRunningOperationResult -operationId '$operationId'" -Level Verbose
+                return [PSCustomObject]@{
+                    Location     = $responseHeader["Location"]
+                    RetryAfter   = $responseHeader["Retry-After"]
+                    OperationId  = $responseHeader["x-ms-operation-id"]
+                }
+            }
+
+            Write-Message -Message "[Test-FabricApiResponse] Operation ID: '$operationId'" -Level Debug
+            Write-Message -Message "[Test-FabricApiResponse] Getting Long Running Operation status" -Level Debug
```

Comment:
this can also. I wont comment on every one but please replace with native commands